aten/src/ATen/core/List_inl.h

```diff
@@ -178,6 +178,21 @@ void ListPtr<T>::push_back(T&& value) const {
   impl_->list.push_back(detail::list_element_from<T, StorageT>(std::move(value)));
 }
 
+template<class T>
+void ListPtr<T>::append(ListPtr<T> b) const {
+  size_type neededSize = this->size() + b.size();
+  if (impl_->list.capacity() < neededSize) {
+    this->reserve(std::max(impl_->list.capacity() * 2, neededSize));
```

Comment:
Don't do this manually. Just ignore `capacity`/`reserve` here and use `std::vector` bulk insertion, which will do the right thing:

    impl_->list.insert(impl_->list.end(), b.impl_->list.begin(), b.impl_->list.end())

or for moving

    impl_->list.insert(v1.end(), make_move_iterator(b.impl_->list.begin()), make_move_iterator(b.impl_->list.end()));